concepts/classes/about.md

```diff
@@ -0,0 +1,260 @@
+Every instance object includes a hidden, internal property referred to as `[[prototype]]` in the language specification.
+It holds a reference to the value of the `prototype` key of the constructor function.
+Yes, you read that correctly, a JavaScript function can have key/value pairs because it is also an object behind the scenes.
+
+Since 2015, `[[prototype]]` can be accessed via `Object.getPrototypeOf()`.
+Before that, it was accessible via the key `__proto__` in many environments.
+
+Do not confuse the prototype of an object (`[[prototype]]`) with the `prototype` property of the constructor function.
+
+### Instance Fields
+
+Often, you want all the derived objects to include some fields and pass some initial values for those when the object is constructed.
+This is facilitated via the [`this` keyword][mdn-this].
+Inside the constructor function, `this` represents the new object that will be created via `new`.
+`this` is automatically returned from the constructor function when it is called with `new`.
+
+That means we can add fields to the new instance by adding them to `this` in the constructor function.
+
+```javascript
+function Car(color, weight) {
+  this.color = color;
+  this.weight = weight;
+  this.engineRunning = false;
+}
+
+const myCar = new Car('red', '2mt');
+myCar.color;
+// => 'red'
+myCar.engineRunning;
+// => false
+```
+
+### Instance Methods
+
+Methods are added via the `prototype` property of the constructor function.
+Inside a method, you can access the fields of the instance via `this`.
+This works because of the following general rule.
+
+> When a function is called as a method of an object, its `this` is set to the object the method is called on. [^1]
+
+```javascript
+function Car() {
+  this.engineRunning = false;
+  // ...
+}
+
+Car.prototype.startEngine = function () {
+  this.engineRunning = true;
+};
+
+Car.prototype.addGas = function (litre) {
+  // ...
+};
+
+const myCar = new Car();
+myCar.startEngine();
+myCar.engineRunning;
+// => true
+```
+
+### The Prototype Chain
+
+`myCar` in the example above is a regular JavaScript object and if we would inspect it (e.g. in the browser console) we would not find a property `startEngine` with a function as value directly inside the `myCar` object.
+So how does the code above even work then?
+
+The secret here is called the _prototype chain_.
+When you try to access any property (field or method) of an object, JavaScript first checks whether the respective key can be found directly in the object itself.
+If not, it continues to look for the key in the object referenced by the `[[prototype]]` property of the original object.
+As mentioned before, in our example `[[prototype]]` points to the `prototype` property of the constructor function.
+That is where JavaScript would find the `startEngine` function because we added it there.
```

Comment:
I think it makes sense to add the example in code again after this paragraph, to illustrate it.